### DISCUSSION TITLE
en/chapter_sorting/counting_sort/

### DISCUSSION THREAD

**Discussion by giscus** [krahets/hello-algo#1459] [Announcements]
# en/chapter_sorting/counting_sort/

Data Structures and Algorithms Crash Course with Animated Illustrations and Off-the-Shelf Code

https://www.hello-algo.com/en/chapter_sorting/counting_sort/

**Comment by JeninSutradhar** [comment]
C++ implementation is mostly correct but it does not handle the case where the input array is empty.

```java
if (nums == null || nums.length == 0) {
        return; // handle empty array
}
```
